sdk/openai/Azure.AI.OpenAI/src/Custom/Choice.Serialization.cs

```diff
@@ -49,7 +49,7 @@ internal static Choice DeserializeChoice(JsonElement element)
                         continue;
                     }
                 }
-                if (property.NameEquals("finishReason"))
+                if (property.NameEquals("finish_reason"))
```

Comment:
This is probably not very important, but this comparison will be faster if you do:
```csharp
if (property.NameEquals("finish_reason"u8)) ...
```

JsonDocument stores the data as UTF8. If you call the UTF16 taking overload, the overload will convert the UTF16 string to UTF8 on the fly, before actually comparing the strings. But you can just append `u8` to the end of string literals to make them UTF8 at compile time.

Reply:
That's a great observation. We can optimize it in our custom deserialization, but I suspect that having the C# emitter aware of this (for the generated code coming out of cadl) would be even better; I believe even these "custom" deserializes are just lightly modified generated code.

@m-nash, do you know if smart use of UTF8 string literals is already something on the radar for C# emission? Should I file an issue to track the request?

Reply:
Resolving for the purposes of this PR, in any event -- this change was ported from a different PR that's already merged (i.e. this file isn't being changed here now).